powerstore/resource_volume.go

```diff
@@ -242,6 +276,15 @@ func (r resourceVolume) Create(ctx context.Context, req tfsdk.CreateResourceRequ
 		return
 	}
 
+	valid, errmsg := fetchByName(r.p.client, &plan)
```

Comment:
define fetchByName as method on r
in that way you won't need to pass client
and also will keep clean the package as functions won't be scattered around